assets/src/blocks/Media/MediaEditor.js

```diff
@@ -0,0 +1,143 @@
+import { Fragment, useEffect, useState, useCallback } from "@wordpress/element";
+import { PanelBody } from '@wordpress/components';
+import { MediaPlaceholder } from "@wordpress/editor";
+import { InspectorControls } from '@wordpress/block-editor';
+
+import { URLInput } from "../../components/URLInput/URLInput";
+import { MediaEmbedPreview } from "./MediaEmbedPreview";
+import { MediaElementVideo } from './MediaElementVideo';
+import { useSelect } from '@wordpress/data';
+import { debounce } from 'lodash';
+
+const { __ } = wp.i18n;
+const { RichText } = wp.blockEditor;
+
+const MediaInspectorOptions = ({ attributes, setAttributes }) => {
+  // Using a state to prevent the input losing the cursor position, a React issue reported multiple times
```

Comment:
Using a cached state locally (instead of the attribute directly) prevented this issue: 
https://github.com/facebook/react/issues/14904

I haven't tried turning it into an uncontrolled component and using `defaultValue` as stated in the last comment of the issue, it is a bit of a hack. Uncontrolled components are not totally ruled out by React, but I'd say they are mildly discouraged. 

I couldn't trace why it happens when you set the attribute directly, using a local state fixed it.

Reply:
See also: https://github.com/facebook/react/issues/955#issuecomment-469352730